open-metadata-implementation/adapters/open-connectors/governance-daemon-connectors/view-generator-connectors/src/main/java/org/odpi/openmetadata/openconnectors/governancedaemonconnectors/viewgenerator/derby/DerbyConnector.java

```diff
@@ -0,0 +1,463 @@
+                    null,
+                    auditCode.getSystemAction(),
+                    auditCode.getUserAction());
+        }
+    }
+
+    /**
+     * Delete a table
+     *
+     * @param tableName table name
+     * @return boolean whether the table is deleted successfully
+     */
+    @Override
+    public boolean deleteLogicalTable(String tableName) {
+        final String actionDescription = "deleteLogicalTable";
+
+        try {
+            derbyStatement = derbyConnection.createStatement();
+            derbyStatement.setQueryTimeout(timeoutInSecond);
+            derbyStatement.executeQuery("call removelt('" + tableName + "')");
+
+            return true;
+
+        } catch (SQLException e) {
+            log.error("Error deleting table", e);
+            if (omrsAuditLog != null) {
+                auditCode = DerbyConnectorAuditCode.CONNERCTOR_QUERY_ERROR;
+                omrsAuditLog.logRecord(actionDescription,
+                        auditCode.getLogMessageId(),
+                        auditCode.getSeverity(),
+                        auditCode.getFormattedLogMessage(),
+                        null,
+                        auditCode.getSystemAction(),
+                        auditCode.getUserAction());
+            }
+            return false;
+        }
+    }
+
+    @Override
+    public List<LogicTable> getAllLogicTables() {
+        final String actionDescription = "getAllLogicTables";
+
+        List logicTableList = new ArrayList();
+
+        try {
+            derbyStatement = derbyConnection.createStatement();
+            derbyStatement.setQueryTimeout(timeoutInSecond);
+            ResultSet resultSet = derbyStatement.executeQuery(getLogicTablesQuery);
+            while (resultSet.next()) {
+                logicTableList.add(extractLogicTableDefinition(resultSet));
+            }
+
+        } catch (SQLException e) {
+            log.error("Error in getting all the logic tables: ", e);
+            if (omrsAuditLog != null) {
+                auditCode = DerbyConnectorAuditCode.CONNERCTOR_QUERY_ERROR;
+                omrsAuditLog.logRecord(actionDescription,
+                        auditCode.getLogMessageId(),
+                        auditCode.getSeverity(),
+                        auditCode.getFormattedLogMessage(),
+                        null,
+                        auditCode.getSystemAction(),
+                        auditCode.getUserAction());
+            }
+        }
+        return logicTableList;
+    }
+
+
+    public boolean executeCustomizedQuery(String query) {
+
+        final String actionDescription = "executeCustomizedQuery: " + query;
+
+        try {
+            derbyStatement = derbyConnection.createStatement();
+            derbyStatement.setQueryTimeout(timeoutInSecond);
+            derbyStatement.executeQuery(query);
+            return true;
+        } catch (SQLException e) {
+            log.error("Error deleting table", e);
```

Comment:
I would be useful to have a meaningful error message, the query here is not only for delete